releasenotes/notes/0.22/add-gradients-with-primitives-561cf9cf75a7ccb8.yaml

```
@@ -0,0 +1,30 @@
+---
+features:
+  - |
+    Added a new module :mod:`qiskit.algorithms.gradient` that contains
+    classes which are used to compute gradients using the primitive
+    interfaces defined in :mod:`qiskit.primitives`. There are 4 types of
+    gradient classes: Finite Difference, Parameter Shift, Linear
+    Combination of Unitary, and SPSA with implementations that either use
+    an instance of the :class:`~.BaseEstimator` interface:
+
+      * :class:`~.ParamShiftEstimatorGradient`
+      * :class:`~.LinCombEstimatorGradient`
+      * :class:`~.FiniteDiffEstimatorGradient`
+      * :class:`~.SPSAEstimatorGradient`
+
+    or an instance of the :class:`~.BaseSampler` interface:
+
+      * :class:`~.ParamShiftSamplerGradient`
+      * :class:`~.LinCombSamplerGradient`
+      * :class:`~.FiniteDiffSamplerGradient`
+      * :class:`~.SPSASamplerGradient`
```

Might be nice to be a bit more precise why these two types exist:
```suggestion
      * :class:`~.SPSASamplerGradient`

    The estimator-based gradients compute the gradient of expectation 
    values, while the sampler-based gradients return gradients of the
    measurement outcomes (also referred to as "probability gradients").
```